Juice doesn't compile without native

Hi, the docs say I should be able to enable the `cuda` feature without the `native` feature to maximize functionality:
```toml
juice = { version = "0.2.5", default-features = false, features = ["cuda"] }
```

But it fails to compile:

```Rust
error[E0412]: cannot find type `Native` in this scope
   --> C:\Users\dkolsoi\.cargo\registry\src\github.com-1ecc6299db9ec823\juice-0.2.5\src\layer.rs:785:40
    |
785 |         let native_backend = Backend::<Native>::default().unwrap();
    |                                        ^^^^^^ not found in this scope
    |
help: consider importing this struct
    |
6   | use coaster::frameworks::Native;
    |

error[E0412]: cannot find type `Native` in this scope
   --> C:\Users\dkolsoi\.cargo\registry\src\github.com-1ecc6299db9ec823\juice-0.2.5\src\layer.rs:920:44
    |
920 |             let native_backend = Backend::<Native>::default().unwrap();
    |                                            ^^^^^^ not found in this scope
    |
help: consider importing this struct
    |
6   | use coaster::frameworks::Native;
    |

error[E0412]: cannot find type `Native` in this scope
  --> C:\Users\dkolsoi\.cargo\registry\src\github.com-1ecc6299db9ec823\juice-0.2.5\src\util.rs:16:36
   |
16 | pub fn native_backend() -> Backend<Native> {
   |                                    ^^^^^^ not found in this scope
   |
help: consider importing this struct
   |
3  | use coaster::frameworks::Native;
   |

error[E0433]: failed to resolve: use of undeclared type or module `Native`
  --> C:\Users\dkolsoi\.cargo\registry\src\github.com-1ecc6299db9ec823\juice-0.2.5\src\util.rs:17:21
   |
17 |     let framework = Native::new();
   |                     ^^^^^^ not found in this scope
   |
help: consider importing this struct
   |
3  | use coaster::frameworks::Native;
   |
```

I'm using Win10 + Rust v1.46

--- a/juice/FEATURE-FLAGS.md
+++ b/juice/FEATURE-FLAGS.md
@@ -8,14 +8,14 @@ Optimally we would like to always have to choice of running Juice on all backend
 However in reality there are some tradeoffs that have to be made.
 
 One problem is that certain backends require the presence of special hardware to
-run (CUDA needs NVIDIA GPUs), or the libraries to address them are not present on
+run (CUDA needs NVIDIA GPUs and cuda drivers and libraries on the host, OpenCL requires a capable CPU/GPU and runtime libs), or the libraries to address them are not present on
 the developers machine which is necessary for compilation.
 
 Another challenge is that not all backends have support for the same operations,
 which constrains neural networks with special requirements to the backends that
-provide those operations. Due to some limitations in the current version of Rust
-(1.7) allowing differently featured backends can not be that easily supported.
-See [Issue #7](https://github.com/spearow/juice/issues/7).
+provide those operations.
+
+Tracking issue for implementing a specialization based system is in [Issue #7](https://github.com/spearow/juice/issues/7).
 
 ## The solution
 
@@ -27,30 +27,16 @@ Luckily, Cargo, Rust's package manager has built-in support for feature flags.
 A simple dependency with additional features enabled in a `Cargo.toml` looks like this:
 ```toml
 [dependencies]
-juice = { version = "0.2.2", features = ["cuda"] }
+juice = { version = "0.2", features = ["native", "cuda"] }
 ```
 
-Feature flags are usually used in an additive way, but **some configurations
-of features for Juice might actually take away some functionality**.
-We do this because we want the models to be portable across different backends,
-which is not possible if e.g. the CUDA backend supports Convolution layers while
-the Native backend doesn't. To make it possible we deactivate those features that
-are only available on a single backend, effectively "dumbing down" the backends.
+Feature flags are usually used in an additive way, and Juice is no exception.
 
-Example:
-- feature flags are `cuda` -> `Convolution` Layer **is available** since the CUDA backend provides the required traits and there is no native backend it has to be compatible with.
-- feature flags are `native` -> `Convolution` Layer **is not available** since the native backend does not provide the required traits and there are no other frameworks present.
-- feature flags are `native cuda` -> `Convolution` Layer **is not available** since the native backend does not provide the required traits, and the CUDA backend has been dumbed down.
+Some individual backends might not implement all features, i.e. currently the `native` backend has an incomplete implementation of the `Convolution` Layer, as such it will panic on use. The `cuda` backend has a complete `Convolution` Layer, which **is available** since the CUDA backend implements the required traits.
 
-## Using the feature flags
+Note that the `native` backend is always required for the time being, in order to load data to `opencl` and `cuda` frameworks.
 
-One thing we have ignored until now are default feature flags. Cargo allows to
-define a set of features that should be included in a package by default .
-One of the default feature flags of Juice is the `native` flag. When looking at
-the above example you might notice that the only way we can unleash the full
-power of the CUDA backend is by deactivating the default `native` flag.
-Cargo allows us to do that either via the `--no-default-features` on the CLI or
-by specifying `default-feature = false` for a dependency in `Cargo.toml`.
+The default features set is `cuda native`.
 
 ### In your project
 
@@ -62,9 +48,9 @@ A typical example (including coaster) would look like this:
 
 ```toml
 [dependencies]
-juice = { version = "0.2.2", default-features = false }
+juice = { version = "0.2", default-features = false }
 # the native coaster feature is neccesary to read/write tensors
-coaster = { version = "0.1.0", default-features = false, features = ["native"] }
+coaster = { version = "0.1", default-features = false, features = ["native"] }
 
 [features]
 default = ["native"]

